### DISCUSSION TITLE
Set host for every route in one place?

### DISCUSSION THREAD

**Discussion by SuperSajuuk** [pallets/flask#4235] [Q&A] [answered]
Hiya. I'm doing a bit of cleanup of my code, and one major thing I need to fix is the fact that I define a host for all my routes under host_matching. Currently, my project has multiple views [ie Blueprints, the folder is just called "views"], but every one of them manually imports a variable from a config file to set the "host" that the route is bound to. Is there a way to set that globally in the whole flask app when creating the Flask object so it doesn't need to be set on each individual route?

I noticed a "static_host" option, but this appears to be for the routes involving static files [ie "static/something.txt"], and not "all routes in the entire app". Thanks for the response in advance!

**Comment by davidism** [comment, marked as answer]
If all routes are for one host, you don't need to use host matching.

You can use `functools.partial`, or otherwise wtite a decorator, to wrap `app.route` and automatically pass the value you want.

**Reply by SuperSajuuk** [reply]
Hi @davidism tthanks for the reply. Could you show a simple example so I can better understand what you mean? I can't quite visualise how that would look in practice :D

**Reply by tachyondecay** [reply]
It would be helpful if you could provide us with the code you’re using so we can better understand what you’re attempting to do.

**Reply by SuperSajuuk** [reply]
Hi @tachyondecay sure that makes sense. Here's an example snippet of code [with stuff redacted for security]. Note that I make heavy use of blueprints in my project, hence the "Auth" part as that represents a blueprint of my Flask app:
```py
# Discord Login Route
@Auth.route('/login', host=config.shost)
def login_via_discord():
	if session.get('uid') is not None:
		flash("You've already signed into the dashboard, so you were automatically logged in.", "success")
		return redirect(url_for('root.dashboard_home'))

	# Request authorisation.
	# If there's a particular parameter in this route,
	# set a cookie so we can get it later in the flow.
	# For security reasons, this has been redacted.
				
	scope = request.args.get(
		'scope',
		'identify guilds.join')
	discord = make_session(scope=scope.split(' '))
	authorization_url, state = discord.authorization_url(AUTHORIZATION_BASE_URL)
	session['oauth2_state'] = state
	return redirect(authorization_url+"&prompt=none")
```
As you can see, in the route() decorator, I define host=config.shost. config.shost refers to a variable that defines the host this route is bound to. All the routes in my project use the same host=config.shost definition, and in the home.py file [where Flask is init'd]:
```py
# Create the application and configure it
core = Flask(__name__)
Mobility(core)
core.config['REDIS_URL'] = "path/to/redis/url/and/db"
redis_client = FlaskRedis(core)
core.url_map.host_matching = True
core.request_class.user_agent_class = ParsedUserAgent
core.config['MYSQL_DATABASE_CHARSET'] = "utf8mb4"
core.config['SESSION_COOKIE_SECURE'] = True
core.config['SECRET_KEY'] = config.OAUTH2_CLIENT_SECRET
core.config['MAX_CONTENT_LENGTH'] = 4 * 1024 * 1024
core.secret_key = sess_secret

# Register Blueprints
core.register_blueprint(Auth)
# several others are registered here but not relevant to discussion
```
However, the host is always the same in every route thats made, so if there's a way to "not" need to define the host of every route, since the app is only for a single host url, that would be great.

**Reply by ThiefMaster** [reply]
Yeah, you should not set the host on the route level at all in almost all cases - especially if you use the same one everywhere. Canonicalizing the host (redirecting/rejecting requests made for a different host) should be done on the web server level anyway (just like enforcing TLS).

**Reply by SuperSajuuk** [reply]
> Yeah, you should not set the host on the route level at all in almost all cases - especially if you use the same one everywhere. Canonicalizing the host (redirecting/rejecting requests made for a different host) should be done on the web server level anyway (just like enforcing TLS).

@ThiefMaster I see, thanks for the guidance! I already use a virtualhost anyway to proxy apache to Gunicorn [which is how the dashboard is run in the production environment], so I can just add something there to ensure it only responds to requests from the specified host.
